packages/teams-js/src/public/runtime.ts

```diff
@@ -439,6 +439,42 @@ export const mapTeamsVersionToSupportedCapabilities: Record<string, Array<ICapab
 };
 
 const generateBackCompatRuntimeConfigLogger = runtimeLogger.extend('generateBackCompatRuntimeConfig');
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * Merges the capabilities of two runtime objects. Fully supports arbitrarily nested capabilities/subcapabilities.
+ *
+ * Note that the two parameters are named "baselineRuntime" and "runtimeToMergeIntoBaseline" to make it easier
+ * to understand what this function is doing -- but the order of the parameters does not matter, it's just merging
+ * the two runtime objects together.
+ *
+ * Also note that this function isn't actually doing anything specific to capabilities/runtime. It's just doing a
+ * generic merge of two objects.
+ *
+ * @param baselineRuntime the baseline runtime object
+ * @param runtimeToMergeIntoBaseline the runtime object to merge into the baseline
+ * @returns the merged runtime object
+ */
+function mergeRuntimeCapabilities(baselineRuntime: object, runtimeToMergeIntoBaseline: object): object {
+  const merged: object = { ...baselineRuntime };
+
+  for (const key in runtimeToMergeIntoBaseline) {
+    if (Object.prototype.hasOwnProperty.call(runtimeToMergeIntoBaseline, key)) {
+      if (typeof runtimeToMergeIntoBaseline[key] === 'object' && !Array.isArray(runtimeToMergeIntoBaseline[key])) {
+        merged[key] = mergeRuntimeCapabilities(baselineRuntime[key] || {}, runtimeToMergeIntoBaseline[key]);
+      } else {
+        if (!(key in baselineRuntime)) {
+          merged[key] = runtimeToMergeIntoBaseline[key];
```

Comment:
The doc said that the order of the parameters does not matter, but I think this will make it so merged[key] is determined by whatever is in baselineRuntime if both baselineRuntime and runtimeToMergeIntoBaseline have the same key.

Reply:
Yes, although this only applies for properties that are not objects (e.g., strings), which shouldn't be in the runtime anyway. I've added a comment to indicate this.

Reply:
Isn't it possible that it is an object since something that would be `typeof runtimeToMergeIntoBaseline[key] === 'object'` but not `!Array.isArray(runtimeToMergeIntoBaseline[key]` would go into the `else`?

Reply:
The only two types that have `typeof === object` are actual objects and arrays.

This runtime should never have anything but objects (actual objects) in it. Many things throughout the code would break if arrays or strings or anything else were in here.

As far as I can tell, in order for us to get into the else case, it means that either it's not an object (`typeof runtimeToMergeIntoBaseline[key] !== 'object'`) or it is an "object" but it's an Array (`Array.isArray(runtimeToMergeIntoBaseline[key])`)

Does this address your concern?